compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryCustom.java

```diff
@@ -0,0 +1,111 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package compute.disks;
+
+//[START compute_disk_create_secondary_custom]
+import com.google.cloud.compute.v1.Disk;
+import com.google.cloud.compute.v1.DiskAsyncReplication;
+import com.google.cloud.compute.v1.DisksClient;
+import com.google.cloud.compute.v1.GuestOsFeature;
+import com.google.cloud.compute.v1.Operation;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class CreateDiskSecondaryCustom {
+  public static void main(String[] args)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // TODO(developer): Replace these variables before running the sample.
+    // The project that contains the primary disk.
+    String primaryProjectId = "PRIMARY_PROJECT_ID";
+    // The project that contains the secondary disk.
+    String secondaryProjectId = "SECONDARY_PROJECT_ID";
+    // Name of the primary disk you want to use.
+    String primaryDiskName = "PRIMARY_DISK_NAME";
+    // Name of the zone in which your primary disk is located.
+    // Learn more about zones and regions:
+    // https://cloud.google.com/compute/docs/disks/async-pd/about#supported_region_pairs
+    String primaryDiskZone = "us-central1-a";
+    // Name of the disk you want to create.
+    String secondaryDiskName = "SECONDARY_DISK_NAME";
+    // Name of the zone in which you want to create the secondary disk.
+    String secondaryDiskZone = "us-east1-c";
+    // Size of the new disk in gigabytes.
+    long diskSizeGb = 30L;
+    // The type of the disk you want to create. This value uses the following format:
+    // "projects/{projectId}/zones/{zone}/diskTypes/
+    // (pd-standard|pd-ssd|pd-balanced|pd-extreme)".
+    String diskType = String.format(
+        "projects/%s/zones/%s/diskTypes/pd-balanced", secondaryProjectId, secondaryDiskZone);
+
+    createDiskSecondaryCustom(primaryProjectId, secondaryProjectId, primaryDiskName,
+        secondaryDiskName, primaryDiskZone, secondaryDiskZone, diskSizeGb,  diskType);
+  }
+
+  // Creates a secondary disk with specified custom parameters.
+  public static Disk createDiskSecondaryCustom(String primaryProjectId, String secondaryProjectId,
+      String primaryDiskName, String secondaryDiskName, String primaryDiskZone,
+      String secondaryDiskZone, long diskSizeGb,  String diskType)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    String primaryDiskSource = String.format("projects/%s/zones/%s/disks/%s",
+        primaryProjectId, primaryDiskZone, primaryDiskName);
+
+    DiskAsyncReplication asyncReplication = DiskAsyncReplication.newBuilder()
```

Comment:
Please move all compute API related code within the try catch. Applies to all samples across PRs.